sklearn/utils/validation.py

```diff
@@ -1655,6 +1655,51 @@ def check_is_fitted(estimator, attributes=None, *, msg=None, all_or_any=all):
         raise NotFittedError(msg % {"name": type(estimator).__name__})
 
 
+def _estimator_has(attr, delegates=None):
+    """Check if we can delegate a method to the underlying estimator.
+
+    We check the `delegates` in the order they are passed.
+    By default, we first check the fitted estimator if available,
+    otherwise we check the unfitted estimator.
+
+    getattr() returns True if the attribute exists in the object,
+    and raises `AttributeError` if it doesn't.
+
+    If none of the delegates are present in the object, ValueError is raised.
+
+    Parameters
+    ----------
+    attr : str
+        Name of the attribute the delegate might or might not have.
+
+    delegates: list
+        List of sub-estimator(s).
+
+    Returns
+    -------
+    check : function
+        Function to check if the delegate has the attribute.
+    """
+
+    if delegates is None:
+        delegates = ["estimator_", "estimator"]
+
+    def check(self):
+        for delegate in delegates:
+            if hasattr(self, delegate):
+                # In meta estimators with multiple sub estimators,
+                # only the attribute of the first sub estimator is checked,
+                # assuming uniformity across all sub estimators.
+                if delegate == "estimators_":
```

Comment:
I don't like this statement here. Basically, we could do the same for `estimators` as well. But then, we might have third party that have something different and then they cannot use this tool. For instance, in imbalanced-learn, we would use this pattern with `samplers_` and `samplers` for instance.

So I would be a bit more lenient here and only check if the attribute can be iterated:

```python
if hasattr(self, delegate):
    delagator = getattr(self, delegate)
    if isinstance(delegator, Sequence):
        return getattr(delegator[0], attr)
    else:
        return getattr(delegator, attr)
```

where `Sequence` can be imported from `collections.abc`